javascript/jsx/JavaScriptParser.g4

```diff
@@ -0,0 +1,557 @@
+    | <assoc=right> singleExpression '**' singleExpression                  # PowerExpression
+    | singleExpression ('*' | '/' | '%') singleExpression                   # MultiplicativeExpression
+    | singleExpression ('+' | '-') singleExpression                         # AdditiveExpression
+    | singleExpression '??' singleExpression                                # CoalesceExpression
+    | singleExpression ('<<' | '>>' | '>>>') singleExpression               # BitShiftExpression
+    | singleExpression ('<' | '>' | '<=' | '>=') singleExpression           # RelationalExpression
+    | singleExpression Instanceof singleExpression                          # InstanceofExpression
+    | singleExpression In singleExpression                                  # InExpression
+    | singleExpression ('==' | '!=' | '===' | '!==') singleExpression       # EqualityExpression
+    | singleExpression '&' singleExpression                                 # BitAndExpression
+    | singleExpression '^' singleExpression                                 # BitXOrExpression
+    | singleExpression '|' singleExpression                                 # BitOrExpression
+    | singleExpression '&&' singleExpression                                # LogicalAndExpression
+    | singleExpression '||' singleExpression                                # LogicalOrExpression
+    | singleExpression '?' singleExpression ':' singleExpression            # TernaryExpression
+    | <assoc=right> singleExpression '=' singleExpression                   # AssignmentExpression
+    | <assoc=right> singleExpression assignmentOperator singleExpression    # AssignmentOperatorExpression
+    | Import '(' singleExpression ')'                                       # ImportExpression
+    | singleExpression TemplateStringLiteral                                # TemplateStringExpression  // ECMAScript 6
+    | yieldStatement                                                        # YieldExpression // ECMAScript 6
+    | This                                                                  # ThisExpression
+    | identifier                                                            # IdentifierExpression
+    | Super                                                                 # SuperExpression
+    | literal                                                               # LiteralExpression
+    | arrayLiteral                                                          # ArrayLiteralExpression
+    | objectLiteral                                                         # ObjectLiteralExpression
+    | htmlElements                                                          # htmlElementExpression
+    | '(' expressionSequence ')'                                            # ParenthesizedExpression
+    ;
+
+htmlElements
+    : htmlElement+
+    ;
+
+htmlElement
+    : '<' htmlTagName htmlAttribute* '>' htmlContent '<''/' htmlTagName '>'
```

Comment:
How if 
```jsx
var a = <h1>title</p>
```

Reply:
Good Point. Not Sure how to handle that. I certainly need help with that.

Reply:
I think it's somehow tolerable. XML grammar seems has same problem. It can check by other part of compiler.

My solution is push tag name into stack at start(use action), and check it at the end of tag. Not sure it's a good idea, maybe it will cause some performance impact.

Reply:
https://github.com/antlr/grammars-v4/pull/1672/commits/2d7e637155e29df3dd1fe77784de781e9933ce77 Fixes this. the action methods should be added for other platforms.

Reply:
I think performance overhead is negligible.